backend/lib/endpoints/feedback.js

```diff
@@ -0,0 +1,36 @@
+const httpErrors = require("http-errors");
+const { createFeedbackSchema } = require("./schema/feedback");
+
+/*
+ * /api/feedback
+ */
+async function routes(app) {
+  const Feedback = app.mongo.model("Feedback");
+
+  app.post("/", { schema: createFeedbackSchema }, async (req) => {
+    const { userId } = req.body;
+    let ip = "undefined";
+    if (userId) {
+      const userFeedback = await Feedback.find({ userId });
+      if (userFeedback[0]) {
+        return new httpErrors.Conflict("User already submitted feedback");
+      }
+    }
+
+    if (req.ip !== null && typeof req.ip !== "undefined") {
+      ip = req.ip;
+      const feedbackFromIp = await Feedback.find({ ipAddress: ip });
+      if (feedbackFromIp[0]) {
+        return new httpErrors.Conflict(
+          "Feedback already submitted with this IP address",
+        );
+      }
+    }
```

Comment:
What would happen if there was no IP on the request?
We would end up saving it as `undefined` isn't it?
Maybe missing the `else ` case here?

Reply:
Also, adding up to that, the Feedback is required to have an IP address or an authenticated User (from which we can get the location)

Reply:
If there is no ip then it will be undefined, correct. I was unsure of what a desirable name would be otherwise. I don't need an else, because of the fact that I create ip before this if statement.